fuel-core-interfaces/src/relayer.rs

```diff
@@ -64,7 +64,7 @@ pub trait RelayerDb:
     /// Query delegate index to find list of blocks that delegation changed
     /// iterate over list of indexed to find height that is less but closest to da_height
     /// Query that block StakeDiff to find actual delegation change.
-    async fn get_first_lesser_delegation(&mut self,delegate: &Address, da_height: DaBlockHeight) ->  Option<HashMap<Address,u64>> {
+    async fn get_first_lesser_delegation(&mut self,delegate: &Address, da_height: DaBlockHeight) ->  Option<HashMap<ValidatorId,ValidatorStake>> {
```

Comment:
Also unrelated to the PR, but we are forcing the user to reallocate a new map when we can trivially return an iterator.

Reply:
One of the reasons is this: https://github.com/FuelLabs/fuel-core/issues/430
And I think I already tried it but don't remember what other problem was there, I think it is returning impl iter from trait.
I can agree that iterator seems a better and cleaner way, but this is good enough as we are always in need of a full set.